### PR TITLE
PR #779 plus more warning suppression and compile error fix

### PR DESCRIPTION
**Pull request by tandasat**
3730e20 and bc954d0 were added on the top of PR #779

### CODE REVIEW

File: arch/X86/X86Mapping.c [hunk 1]
```diff
@@ -2923,7 +2923,7 @@ static struct insn_reg2 insn_regs_intel2[] = {
 
 static struct insn_reg insn_regs_intel_sorted [ARR_SIZE(insn_regs_intel)];
 
-static int regs_cmp(const void *a, const void *b)
+static int CAPSTONE_API regs_cmp(const void *a, const void *b)
```
Comment (aquynh) [inline arch/X86/X86Mapping.c]: why do we need CAPSTONE_API here now? what was wrong before, without it?

Comment (tandasat) [inline arch/X86/X86Mapping.c]: I explained it in the commit log, please check it out. I should have mentioned the note in the PR message.

Comment (aquynh) [inline arch/X86/X86Mapping.c]: i see. would be better to put that comment in the source code as well?

Comment (tandasat) [inline arch/X86/X86Mapping.c]: Good idea as the function is not a part of API. I will do it shortly.

**Comment by aquynh** [comment]
merged, thanks!

**Pull request merged**